src/core/search/ast_expr.h

```diff
@@ -25,18 +26,17 @@ struct AstStarNode {};
 // Matches all documents where this field has a non-null value
 struct AstStarFieldNode {};
 
-// Matches terms in text fields
-struct AstTermNode {
-  explicit AstTermNode(std::string term);
+template <TagType T> struct AstAffixNode {
+  explicit AstAffixNode(std::string affix) : affix{std::move(affix)} {
+  }
 
-  std::string term;
+  std::string affix;
 };
 
-struct AstPrefixNode {
-  explicit AstPrefixNode(std::string prefix);
-
-  std::string prefix;
-};
+using AstTermNode = AstAffixNode<TagType::REGULAR>;
```

Comment:
while this indeed reduces code duplication, why do you prefer creating 4 differrent types over a single type with an enum? (just curious, no need to change)